crates/sui-graphql-client/src/lib.rs

```diff
@@ -494,20 +492,30 @@ impl Client {
     }
 
     /// Get a page of [`CheckpointSummary`] for the provided parameters.
-    pub async fn checkpoints(
+    pub async fn checkpoints<'a>(
         &self,
-        after: Option<&str>,
-        before: Option<&str>,
-        first: Option<i32>,
-        last: Option<i32>,
+        pagination_filter: PaginationFilter<'a>,
     ) -> Result<Option<Page<CheckpointSummary>>, Error> {
+        let (after, before, first, last) = match pagination_filter.direction {
+            Direction::Forward => (
+                pagination_filter.cursor,
+                None,
+                pagination_filter.limit,
+                None,
+            ),
+            Direction::Backward => (
+                None,
+                pagination_filter.cursor,
+                None,
+                pagination_filter.limit,
+            ),
+        };
```

Comment:
can we factor this out into a private method on PaginationFilter do do this so that it isn't copy-pasted everywhere?

Reply:
Definitely, don't know why I left it like that 🤦

Reply:
Done